### PR TITLE
Change the way calls are created for help requests

### PR DESCRIPTION
**Pull request by LBHSPreston**
# What
- Utilising ef core navigation properties to link a call to a help request
- Added an entity test helper class to be able to customise how objects are created for tests
- Check help request exists before creating a call id attached to it

# Why
- To add confidence that we are adding the call record to the correct help request by searching for the help request before attaching a call to it.  This also helps prevent an error occurring if a call id is passed that does not exist

- Test helper methods were also added to make it easier to customise creating test objects in one place

# Screenshots
- None

# Link to ticket
https://trello.com/c/KFlXvf6f/56-as-an-outbound-shielding-caller-i-need-to-record-each-time-i-make-a-call-and-the-action-of-that-call

# Notes
- This change also introduces the use of navigation properties to associate help requests to calls.  This would also be useful when getting help requests

### CODE REVIEW

File: cv19ResSupportV3.Tests/V3/E2ETests/CreateHelpRequestCall.cs [hunk 1]
```diff
@@ -41,12 +43,26 @@ public async Task CreateHelpRequestCallReturnsTheCorrectInformation()
             Console.WriteLine("123456");
             Console.WriteLine(content);
```
Comment (maysakanoni) [inline cv19ResSupportV3.Tests/V3/E2ETests/CreateHelpRequestCall.cs:44]: nit pick

File: cv19ResSupportV3.Tests/V3/E2ETests/CreateHelpRequestCall.cs [hunk 2]
```diff
@@ -41,12 +43,26 @@ public async Task CreateHelpRequestCallReturnsTheCorrectInformation()
...
             var stringContent = await content.ReadAsStringAsync().ConfigureAwait(true);
-            var convertedResponse = JsonConvert.DeserializeObject<HelpRequestCallCreateResponse>(stringContent);
+            // var convertedResponse = JsonConvert.DeserializeObject<HelpRequestCallCreateResponse>(stringContent);
```
Comment (maysakanoni) [inline cv19ResSupportV3.Tests/V3/E2ETests/CreateHelpRequestCall.cs:46]: this is a nit pick along with lines 43 and 44

Comment (maysakanoni) [inline cv19ResSupportV3.Tests/V3/E2ETests/CreateHelpRequestCall.cs:46]: nit pick, I think we can remove these lines and change from async to void return type?

Comment (LBHSPreston) [reply cv19ResSupportV3.Tests/V3/E2ETests/CreateHelpRequestCall.cs]: I've removed those lines, thanks.

**Review by maysakanoni — APPROVED** [review]
Nice!

**Pull request merged**